📔  Add more documentation (#69)
EthWorks/useDapp: docs/source/core.rst

useBlockMeta
============

useChainCall
============

> Makes a call to a specific contract and returns value. Hook will cause component to refesh whenever new block is mined and value is changed.
> 
> Calls will be combined into a single multicall across all uses of *useChainCall* and *useChainCalls*.
> 
> It is recommended to use `useContractCall`_ where applicable instead of this method.
> 
> *Parameters*
> 
> - ``call: ChainCall | Falsy`` - a single call, also see `ChainCall`_. A call can be `Falsy`, as it is important to keep the same ordering of hooks even if in given render cycle there might be not enough information to perfom a call.
> 
> 

Reviewer: ```suggestion
Makes a call to a specific contract and returns the value. The hook will cause the component to refresh whenever a new block is mined and the value is changed.

Calls will be combined into a single multicall across all uses of *useChainCall* and *useChainCalls*.

It is recommended to use `useContractCall`_ where applicable instead of this method.

*Parameters*

- ``call: ChainCall | Falsy`` - a single call, also see `ChainCall`_. A call can be `Falsy`, as it is important to keep the same ordering of hooks even if in a given render cycle there might be not enough information to perform a call.

```

useChainCalls
=============

> Makes multiple call to a specific contracts and returns values. Hook will cause component to refesh when values change.

Reviewer: ```suggestion
Makes multiple calls to specific contracts and returns values. The hook will cause the component to refresh when values change.
```

Reviewer: @marekkirejczyk

Calls will be combined into a single multicall across all uses of *useChainCall* and *useChainCalls*.

> It is recommended to use `useContractCall`_ where applicable instead of this methods.

Reviewer: ```suggestion
It is recommended to use `useContractCall`_ where applicable instead of this method.
```

*Parameters*

> - ``calls: ChainCall[]`` - list of calls, also see `ChainCall`_. Calls need to be in the same order accros component renders.

Reviewer: ```suggestion
- ``calls: ChainCall[]`` - list of calls, also see `ChainCall`_. Calls need to be in the same order across component renders.
```

useContractCall
===============

> Makes a call to a specific contract and returns value. Hook will cause component to refesh when new block is mined and return value changes.
> A syntax sugar for `useChainCall`_ that uses ABI, function name and arguments instead of raw data.

Reviewer: ```suggestion
Makes a call to a specific contract and returns the value. The hook will cause the component to refresh when a new block is mined and the return value changes.
A syntax sugar for `useChainCall`_ that uses ABI, function name, and arguments instead of raw data.
```

**Parameters**

- ``calls: ContractCall | Falsy`` - a single call to a contract , also see `ContractCall`_

**Returns**

> - ``any[] | undefined`` - result of a call or undefined if call didn't return yet

Reviewer: ```suggestion
- ``any[] | undefined`` - the result of a call or undefined if call didn't return yet
```

useContractCalls
===============

> Makes calls to specific contracts and returns values. Hook will cause component to refesh when new block is mined and any of return values changes.
> A syntax sugar for `useChainCalls`_ that uses ABI, function name and arguments instead of raw data.

Reviewer: ```suggestion
Makes calls to specific contracts and returns values. The hook will cause the component to refresh when a new block is mined and the return values change.
A syntax sugar for `useChainCalls`_ that uses ABI, function name, and arguments instead of raw data.
```

**Generic parameters**

- ``T`` - type of stored value 

**Parameters**

- ``value: T`` - variable to be debounced
- ``delay: number`` - debounce time - amount of time in ms 

**Returns**

- ``T`` - debounced value

**Example**

.. code-block:: javascript

  const [someValue, setValue] = useState(...) 
  const debouncedValue = useDebounce(value, 1000)
  

useDebouncePair
===============

Debounce a pair of values of types T and U. 
It stores a single value but returns after debounced time unless a new value is assigned before the debounce time elapses, in which case the process restarts.

This function is used for debouncing multicall until enough calls are aggregated.

**Generic parameters**

- ``T`` - type of first stored value 
- ``U`` - type of second stored value  

**Parameters**

- ``first: T`` - first variable to be debounced
- ``second: U`` - second variable to be debounced
- ``delay: number`` - deboune time - amount of time in ms 

**Returns**

- ``[T, U]`` - debounced values

useEtherBalance
===============

Returns ether balance of a given account.

**Parameters**

- ``address: string | Falsy`` - address of an account

**Returns**

> - ``BigNumber`` - balance of the account or *undefined* if not connected to network or address is falsy value

Reviewer: ```suggestion
- ``balance: BigNumber | undefined`` - a balance which is BigNumber of the account or *undefined* if not connected to network or address is a falsy value
```

useMulticallAddress
===================

useTokenBalance
===============

Returns a balance of a given token for a given address.

**Parameters**

- ``address: string | Falsy`` - address of an account
- ``tokenAddress: string | Falsy`` - address of a token contract

**Returns**

> - ``BigNumber | undefined`` - balance or undefined if address or token is *Falsy* or not connected

Reviewer: ```suggestion
- ``balance: BigNumber | undefined`` - a balance which is BigNumber or undefined if address or token is *Falsy* or not connected
```

**pollingInterval**
Polling interval for a new block.

ChainCall
=========

> Represents a single call on the blokchain that can be included in multicall.

Reviewer: ```suggestion
Represents a single call on the blockchain that can be included in multicall.
```
